ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -5031,7 +5038,7 @@ private RelNode genLogicalPlan(QB qb, boolean outerMostQB,
 
       // Build Rel for Constraint checks
       Pair<RelNode, RowResolver> constraintPair =
-          genConstraintFilterLogicalPlan(qb, srcRel, outerNameToPosMap, outerRR);
+          genConstraintFilterLogicalPlan(qb, selPair, outerNameToPosMap, outerRR);
```

Comment:
will this work okay when `selectRel == null`?
previous code was passing `srcRel` which is optionally the previous `srcRel`

Reply:
Went through the code where `selectRel` gets its value and I found that it can not be null:
If is coming from `internalGenSelectLogicalPlan` which can create it with the following way
```
outputRel = genUDTFPlan(genericUDTF, genericUDTFName, udtfTableAlias, udtfColAliases, qb,
...
        RelNode udtf = HiveTableFunctionScan.create(cluster, traitSet, list, rexNode, null, retType,
          null);
outputRel = genSelectRelNode(columnList, outputRR, srcRel);
...
      HiveRelNode selRel = HiveProject.create(

outputRel = new HiveAggregate(cluster, cluster.traitSetOf(HiveRelNode.CONVENTION),
```